lib/cursor.js

```diff
@@ -160,7 +165,63 @@ module.exports = {
         });
       }
     });
+    
+    // Filter by start date, in YYYY-MM-DD format. The event must begin on
+    // or after that day.
+    // Use of this filter cancels the upcoming filter
+    self.addFilter('start', {
+      def: null,
+      safeFor: 'public',
+      finalize: function() {
+        var start = self.get('start');
+
+        if (start === null) {
+          return;
+        }
+
+        self.and({
+          startDate: { $gte: start }
+        });
+      },
+      launder: function(s) {
+        s = self.apos.launder.string(s);
+
+        if (!s.match(/^\d\d\d\d\-\d\d\-\d\d$/)) {
+          return null;
+        }
 
+        return s;
+      }
+    });
+
+    // Filter by end date, in YYYY-MM-DD format. The event must end on or
+    // before that day.
+    // Use of this filter cancels the upcoming filter
+    self.addFilter('end', {
+      def: null,
+      safeFor: 'public',
+      finalize: function() {
+        var end = self.get('end');
+
+        if (end === null) {
+          return;
+        }
+
+        self.and({
+          endDate: { $lte: end }
```

Comment:
What if the event is in progress at the end date? Should it be excluded because it happens to go on for a few more days?